### PR TITLE
DOC-270: Run Vale against changed files only

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes DOC-270

### CODE REVIEW

File: netlify/build [hunk 1]
```diff
@@ -41,10 +43,19 @@ if [[ "$CONTEXT" = "production" ]]; then
 fi;
 
 # Set up Vale, a text linter
-wget -O- https://github.com/errata-ai/vale/releases/download/v1.2.7/vale_1.2.7_Linux_64-bit.tar.gz | tar -xz vale
+wget --quiet -O- https://github.com/errata-ai/vale/releases/download/v1.2.7/vale_1.2.7_Linux_64-bit.tar.gz | tar -xz vale >/dev/null 2>/dev/null
```
Comment (ianjevans) [inline netlify/build:46]: The latest Vale is v2.16. Should we try to upgrade?

File: v22.1/add-constraint.md [hunk 2]
```diff
@@ -19,6 +19,8 @@ The [`DEFAULT`](default-value.html) and [`NOT NULL`](not-null.html) constraints
 
 {% include {{ page.version.version }}/sql/combine-alter-table-commands.md %}
 
+We don't like this file bababooie asdlfkmflk
```
Comment (ianjevans) [inline v22.1/add-constraint.md:22]: Misspellings currently break the build. I get that Warnings are easy to ignore, but I'm worried that such a strict policy will cause more headaches than it solves for new terms we haven't yet added to the exception list.

**Review by ianjevans — COMMENTED** [review]
Looks good. I left a couple review comments.

I also think that Vale errors need to be surfaced in the PR if possible. If we can eliminate the need to go to the Netlify build log easily, we should do that. But that may not be possible.

File: v22.1/add-constraint.md (hunk 2 above)
Comment (nickvigilante) [reply v22.1/add-constraint.md:22]: I'll send out a poll to the team to see what people would want to see in terms of functionality.

**Comment by nickvigilante** [comment]
> Looks good. I left a couple review comments.
> 
> I also think that Vale errors need to be surfaced in the PR if possible. If we can eliminate the need to go to the Netlify build log easily, we should do that. But that may not be possible.

I'm thinking about adding a separate check entirely that just displays the output of Vale and splitting up the output.

File: netlify/build (hunk 1 above)
Comment (nickvigilante) [reply netlify/build]: Done.

File: v22.1/add-constraint.md (hunk 2 above)
Comment (nickvigilante) [reply v22.1/add-constraint.md]: I personally would prefer that avenue, and I can add documentation in the wiki as to how to edit the list of terms.

**Comment by stbof** [comment]
@nickvigilante Could you add a broken link to this PR?

**Comment by stbof** [comment]
> @stbof Broken links is an htmltest thing, but there's also another GitHub action for that 😄 

Actually I forgot, I really want to find out broken links locally, not in the build. It takes too long.

**Review by ianjevans — APPROVED** [review]
Looks great. Having the Vale messages in the changed files is so much better.

**Pull request merged**